compositor_render/src/transformations/web_renderer/shader.rs

```diff
@@ -0,0 +1,199 @@
+            pipeline,
+            textures_bgl,
+            sampler,
+        })
+    }
+
+    pub(super) fn render(
+        &self,
+        params: &wgpu::BindGroup,
+        textures: &[Option<&Texture>],
+        target: &NodeTextureState,
+        pts: Duration,
+        clear_color: Option<wgpu::Color>,
+    ) {
+        let mut texture_views: Vec<&wgpu::TextureView> = textures
+            .iter()
+            .map(|texture| match texture {
+                Some(texture) => &texture.view,
+                None => &self.wgpu_ctx.empty_texture.view,
+            })
+            .collect();
+
+        texture_views.extend(
+            (texture_views.len()..super::MAX_WEB_RENDERER_TEXTURES_COUNT as usize)
+                .map(|_| &self.wgpu_ctx.empty_texture.view),
+        );
+
+        let input_textures_bg =
+            self.wgpu_ctx
+                .device
+                .create_bind_group(&wgpu::BindGroupDescriptor {
+                    label: Some("Web renderer input textures bgl"),
+                    layout: &self.textures_bgl,
+                    entries: &[wgpu::BindGroupEntry {
+                        binding: 0,
+                        resource: wgpu::BindingResource::TextureViewArray(&texture_views),
+                    }],
+                });
+
+        let common_params =
+            CommonShaderParameters::new(pts, textures.len() as u32, target.resolution());
+
+        let mut encoder = self
+            .wgpu_ctx
+            .device
+            .create_command_encoder(&Default::default());
+        let clear_color = clear_color.unwrap_or(wgpu::Color::TRANSPARENT);
+
+        let mut render_plane = |input_id: i32, clear: bool| {
+            let load = match clear {
+                true => wgpu::LoadOp::Clear(clear_color),
+                false => wgpu::LoadOp::Load,
+            };
+
+            let mut render_pass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: None,
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    ops: wgpu::Operations { load, store: true },
+                    view: &target.rgba_texture().texture().view,
+                    resolve_target: None,
+                })],
+                depth_stencil_attachment: None,
+            });
+
+            render_pass.set_pipeline(&self.pipeline);
+
+            render_pass.set_push_constants(
+                ShaderStages::VERTEX_FRAGMENT,
+                0,
+                common_params.push_constant(),
+            );
+
+            render_pass.set_bind_group(0, &input_textures_bg, &[]);
+            render_pass.set_bind_group(1, params, &[]);
+            render_pass.set_bind_group(2, &self.sampler.bind_group, &[]);
+
+            self.wgpu_ctx
+                .plane_cache
+                .plane(input_id)
+                .unwrap()
+                .draw(&mut render_pass);
+        };
+
+        if common_params.texture_count == 0 {
```

Comment:
this does not really make sense for web renderer, you will never have zero textures